### PR TITLE
Remove XCache and ZendDataCache support

### PR DESCRIPTION
**Pull request by mtangoo**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ✔️
| Fixed issues  | #19891

### CODE REVIEW

**Comment by samdark** [comment]
> Dunno why it pulled this one as wincache is untouched!

GPT isn't precise :)

File: docs/guide-fr/caching-data.md [hunk 1]
```diff
@@ -100,9 +100,6 @@ Yii prend en charge un large panel de supports de stockage pour cache. Ce qui su
   (une version de redis égale ou supérieure à 2.6.12 est nécessaire).
 * [[yii\caching\WinCache]]: utilise le [WinCache](https://iis.net/downloads/microsoft/wincache-extension) PHP
   ([voir aussi l'extension](https://www.php.net/manual/fr/book.wincache.php)).
-* [[yii\caching\XCache]] _(deprecated)_: utilise l'extension PHP [XCache](https://en.wikipedia.org/wiki/List_of_PHP_accelerators#XCache).
-* [[yii\caching\ZendDataCache]] _(deprecated)_: utilise le 
-  [cache de données Zend](https://files.zend.com/help/Zend-Server-6/zend-server.htm#data_cache_component.htm)
   en tant que médium de cache sous-jacent.
```
Comment (samdark) [inline docs/guide-fr/caching-data.md:103]: Should be removed.

File: docs/guide-es/caching-data.md [hunk 2]
```diff
@@ -79,8 +79,7 @@ se muestra un listado con los componentes de caché disponibles:
 * [[yii\caching\MemCache]]: utiliza las extensiones de PHP [memcache](https://www.php.net/manual/es/book.memcache.php) y [memcached](https://www.php.net/manual/es/book.memcached.php). Esta opción puede ser considerada como la más rápida cuando la caché es manejada en una aplicación distribuida (ej. con varios servidores, con balance de carga, etc..)
 * [[yii\redis\Cache]]: implementa un componente de caché basado en [Redis](https://redis.io/) que almacenan pares clave-valor (requiere la versión 2.6.12 de redis).
 * [[yii\caching\WinCache]]: utiliza la extensión de PHP [WinCache](https://iis.net/downloads/microsoft/wincache-extension) ([ver también](https://www.php.net/manual/es/book.wincache.php)).
-* [[yii\caching\XCache]] _(deprecated)_: utiliza la extensión de PHP [XCache](https://en.wikipedia.org/wiki/List_of_PHP_accelerators#XCache).
-* [[yii\caching\ZendDataCache]] _(deprecated)_: utiliza [Zend Data Cache](https://files.zend.com/help/Zend-Server-6/zend-server.htm#data_cache_component.htm) como el medio fundamental de caché.
+ como el medio fundamental de caché.
```
Comment (samdark) [inline docs/guide-es/caching-data.md:82]: Should be removed.

File: docs/guide/caching-data.md [hunk 3]
```diff
@@ -130,9 +130,6 @@ Yii supports a wide range of cache storage. The following is a summary:
   (redis version 2.6.12 or higher is required).
 * [[yii\caching\WinCache]]: uses PHP [WinCache](https://iis.net/downloads/microsoft/wincache-extension)
   ([see also](https://www.php.net/manual/en/book.wincache.php)) extension.
-* [[yii\caching\XCache]] _(deprecated)_: uses PHP [XCache](https://en.wikipedia.org/wiki/List_of_PHP_accelerators#XCache) extension.
-* [[yii\caching\ZendDataCache]] _(deprecated)_: uses
-  [Zend Data Cache](https://files.zend.com/help/Zend-Server-6/zend-server.htm#data_cache_component.htm)
   as the underlying caching medium.
```
[the remark this answers is not in the document]
Comment (mtangoo) [reply docs/guide/caching-data.md:133]: How did they slip my off my face ;)
Fix is coming!

**Review by samdark — APPROVED** [review]

**Pull request merged**